lib/job_queue/tests/job_torque_test.cpp

```diff
@@ -49,7 +49,10 @@ void setoption_setalloptions_optionsset() {
   test_assert_true( torque_driver_set_option( driver , TORQUE_SUBMIT_SLEEP , "0.25"));
   test_assert_int_equal( 250000 , torque_driver_get_submit_sleep(driver));
 
-  test_assert_true( torque_driver_set_option( driver , TORQUE_DEBUG_OUTPUT , "/tmp/torqueue_debug.txt"));
+  char tmp_path[] = "/tmp/torque_debug_XXXXXX.txt";
+  mkstemp(tmp_path);
```

Comment:
From documentation: 

> RETURN VALUE
> Upon successful completion, mkstemp() shall return an open file descriptor. Otherwise, -1 shall be returned if no suitable file could be created.

Do we want to have an open file descriptor here? We only need the path, and i suspect this can have unintended side effects.

Reply:
I agree, which is why I started out with `std::tmpnam` instead.

Would adding a close to it, as it is in its current state, suffice? Or would you rather use a different implementation?

Reply:
It's interesting, because the reason why it has to create and open the file is because it wants to guarantee that you don't create collisions with others trying to create temporary files. I don't know how the torque driver works, but if that is just appending to the file it gets, then you could just close the file immediately after you get it back from `mkstemp`. That way the `mkstemp` would serve as a `touch` and then you can use it as normal in the torque driver.

Reply:
We should also handle the error case when `mkstemp` returns `-1`.